src/key_checker.hpp

```diff
@@ -0,0 +1,64 @@
+#pragma once
+#include <algorithm>
+#include "settings.hpp"
+#include "math_utils.hpp"
+#include "system_event.hpp"
+
+namespace PanzerChasm
+{
+
+using KeyCode= SystemEvent::KeyEvent::KeyCode;
+
+class KeyChecker
+{
+public:
+	KeyChecker( Settings& settings, const InputState& input_state )
+		: settings_(settings), input_state_(input_state), event_(SystemEvent()), event_source_(false)
+	{}
+	KeyChecker( Settings& settings, const SystemEvent& event )
+		: settings_(settings), input_state_(InputState()), event_(event), event_source_(true)
+	{}
+
+	bool operator()( const char* const key_setting_name, const KeyCode default_value ) const
```

Comment:
I see no reason to define all methods directly inside the class. Let's move methods bodies into _.cpp_ file, like in other classes in this project.